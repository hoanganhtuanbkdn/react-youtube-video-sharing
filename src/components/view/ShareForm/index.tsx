import React from 'react';

import { Button, Form, Input } from 'antd';

export default function ShareForm({
	onFinish,
	loading,
}: {
	onFinish: (values: any) => void;
	loading: boolean;
}) {
	return (
		<Form
			name="basic"
			labelCol={{ span: 8 }}
			wrapperCol={{ span: 16 }}
			initialValues={{ videoUrl: '' }}
			onFinish={onFinish}
			autoComplete="new-password"
		>
			<Form.Item
				label="Youtube URL"
				name="videoUrl"
				rules={[
					{
						required: true,
						message: 'Please input a Youtube url!',
					},
				]}
			>
				<Input
					id="videoUrl"
					data-testid="videoUrl"
					autoComplete="off"
				/>
			</Form.Item>

			<div className="row w-full justify-end gap-5">
				<Button
					type="primary"
					data-testid="submitShareBtn"
					className="submitShareBtn"
					htmlType="submit"
					loading={loading}
				>
					Share
				</Button>
			</div>
		</Form>
	);
}
