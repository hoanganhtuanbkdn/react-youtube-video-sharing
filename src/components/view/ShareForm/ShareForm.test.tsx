import { render } from '@testing-library/react';
import React from 'react';

import ShareForm from '.';

// Must be imported before the tested file
Object.defineProperty(window, 'matchMedia', {
	writable: true,
	value: jest.fn().mockImplementation((query) => ({
		matches: false,
		media: query,
		onchange: null,
		addListener: jest.fn(), // deprecated
		removeListener: jest.fn(), // deprecated
		addEventListener: jest.fn(),
		removeEventListener: jest.fn(),
		dispatchEvent: jest.fn(),
	})),
});
describe('rendering and submitting share form', () => {
	test('renders ShareForm correctly', () => {
		const handleClick = jest.fn((values: any) => {
			console.log('values', values);
		});
		const { container } = render(
			<ShareForm onFinish={handleClick} loading={false} />
		);

		expect(container.firstChild).toMatchSnapshot();

		// const Input = container.querySelector('#videoUrl');

		// expect(Input).toBeInTheDocument();

		// // fireEvent.change(Input!, {
		// // 	target: { value: 'https://www.youtube.com/watch?v=qgb-bdEEI-M' },
		// // });
		// // console.log(Input);

		// const Button = container.querySelector('.submitShareBtn');

		// expect(Button).toBeInTheDocument();

		// // fireEvent.click(Button!);

		// // expect(handleClick).toHaveBeenCalledTimes(1);
	});
});
