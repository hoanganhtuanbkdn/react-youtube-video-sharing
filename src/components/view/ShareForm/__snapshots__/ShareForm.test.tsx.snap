// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering and submitting share form renders ShareForm correctly 1`] = `
<form
  autocomplete="new-password"
  class="ant-form ant-form-horizontal css-dev-only-do-not-override-w8mnev"
  id="basic"
>
  <div
    class="ant-form-item css-dev-only-do-not-override-w8mnev"
  >
    <div
      class="ant-row ant-form-item-row css-dev-only-do-not-override-w8mnev"
    >
      <div
        class="ant-col ant-col-8 ant-form-item-label css-dev-only-do-not-override-w8mnev"
      >
        <label
          class="ant-form-item-required"
          for="basic_videoUrl"
          title="Youtube URL"
        >
          Youtube URL
        </label>
      </div>
      <div
        class="ant-col ant-col-16 ant-form-item-control css-dev-only-do-not-override-w8mnev"
      >
        <div
          class="ant-form-item-control-input"
        >
          <div
            class="ant-form-item-control-input-content"
          >
            <input
              aria-required="true"
              autocomplete="off"
              class="ant-input css-dev-only-do-not-override-w8mnev"
              data-testid="videoUrl"
              id="videoUrl"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="row w-full justify-end gap-5"
  >
    <button
      class="ant-btn css-dev-only-do-not-override-w8mnev ant-btn-primary submitShareBtn"
      data-testid="submitShareBtn"
      type="submit"
    >
      <span>
        Share
      </span>
    </button>
  </div>
</form>
`;
